scipy/io/tests/test_wavfile.py

```diff
@@ -302,12 +302,42 @@ def test_rifx():
         assert_equal(data1, data2)
 
 
+def test_rf64():
+    # Compare equivalent RF64 and RIFF files
+    for rf64, riff in {('test-44100Hz-le-1ch-4bytes-rf64.wav',
+                        'test-44100Hz-le-1ch-4bytes.wav'),
+                       ('test-8000Hz-le-3ch-5S-24bit-rf64.wav',
+                        'test-8000Hz-le-3ch-5S-24bit.wav')}:
+        rate1, data1 = wavfile.read(datafile(rf64), mmap=False)
+        rate2, data2 = wavfile.read(datafile(riff), mmap=False)
+        assert_equal(rate1, rate2)
+        assert_equal(data1, data2)
+
+
+@pytest.mark.xslow
+def test_write_roundtrip_rf64(tmpdir):
+    dtype = np.dtype("<i8")
+    tmpfile = str(tmpdir.join('temp.wav'))
+    rate = 44100
+    data = np.random.randint(0, 127, (2**29, )).astype(dtype)
+
+    wavfile.write(tmpfile, rate, data)
+
+    rate2, data2 = wavfile.read(tmpfile, mmap=True)
+
+    assert_equal(rate, rate2)
+    assert_(data2.dtype.byteorder in ('<', '=', '|'), msg=data2.dtype)
```

Comment:
Can we have a proper error message here rather than just the datatype? We also have a preference for plain `assert` over `assert_`. So something like

```python
msg = f"... {data2.dtype} ..." # if it won't all fit on one line
assert data2.dtype.byteorder in ('<', '=', '|'), msg
```